### PR TITLE
Medals or citations page 781

### PR DESCRIPTION
**Pull request by imeecuison**
## Description
Medals or citations page in 781.
http://localhost:3001/disability-benefits/apply/form-526-all-claims/disabilities/ptsd-medals-0

## Testing done
tests/config/781.unit.spec.jsx

## Screenshots
screen shot 2018-11-21 at 10 50 45 am

## Acceptance criteria
- [ ] User is presented with instructions regarding the combat decoration details per the Content Document
- [ ] The PTSD Type that the user initially selects (Combat or Non-combat) should be what shows to the user in the header instead of '$PTSDclassification'

- 1)  user selects Combat & Non-combat

- then 'Combat & Non-combat' replaces $PTSDclassification in header of 0781 workflow

- then 'Combat and Non-combat' replaces $PTSDclassification in body of 0781 workflow
User has indicated Combat Decoration(s) was received (yes/no)
- [ ] If yes, then user is able to enter free text for the medal/citation received
- [ ] User is able to proceed to the next step in the process

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/pages/medals.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+export const schema = index => ({
+  type: 'object',
+  properties: {
+    [`medals${index}`]: {
+      type: 'boolean',
+      properties: {},
+    },
+    [`medalType${index}`]: {
+      type: 'string',
```
Comment (robfelder) [inline src/applications/disability-benefits/all-claims/pages/medals.jsx]: Is there supposed to be any type of character limit on this text box?

File: src/applications/disability-benefits/all-claims/content/incidentDate.jsx [hunk 2]
```diff
@@ -3,34 +3,34 @@ import AdditionalInfo from '@department-of-veterans-affairs/formation/Additional
...
-    trouble remembering the exact date you can provide a month and year. You'll
-    have a chance to provide more information later.)
-  </p>
-);
-
-const eventHelpText = (
-  <AdditionalInfo triggerText="What if I can’t remember the date?">
+export const ptsdDateDescription = (
+  <div>
+    <h5>Event date</h5>
```
Comment (robfelder) [inline src/applications/disability-benefits/all-claims/content/incidentDate.jsx]: Is this supposed to be an h5 or an h3? The flow seems inconsistent right now between this page, the medals page, and the Event Description page.

File: src/applications/disability-benefits/all-claims/content/medals.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+import { getPtsdClassification } from './ptsdClassification';
+
+export const MedalsDescription = ({ formData, index, formType }) => {
+  const { incidentText } = getPtsdClassification(formData, formType);
+  const isFirstIncident = index === 0;
+  return (
+    <div>
+      <h5>Medals or Citations</h5>
```
Comment (robfelder) [inline src/applications/disability-benefits/all-claims/content/medals.jsx]: Same here about the h5 vs an h3

**Review by robfelder — CHANGES_REQUESTED** [review]
Overall, things look good. I had a few questions about the header size, and limit on the medal description, and one formatting request :).

File: src/applications/disability-benefits/all-claims/pages/medals.jsx (hunk 1 above)
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/pages/medals.jsx]: There's currently no max lengths set on this per schema.

File: src/applications/disability-benefits/all-claims/pages/index.js [hunk 4]
```diff
@@ -258,6 +258,7 @@ import {
   schema as conclusionAssaultSchema,
 } from './conclusionAssault';
 
+import { uiSchema as medalsUISchema, schema as medalsSchema } from './medals';
```
[the remark this answers is not in the document]
Comment (robfelder) [reply src/applications/disability-benefits/all-claims/pages/index.js]: ohhhhhh, gotcha.

File: src/applications/disability-benefits/all-claims/tests/config/781.unit.spec.js [hunk 5]
```diff
@@ -33,6 +33,17 @@ describe('781/781a incident form config iterators', () => {
       };
       testMultipleIncidentDescriptionProps();
     });
+    it('should return three medals page config obects', () => {
```
[the remark this answers is not in the document]
Comment (robfelder) [reply src/applications/disability-benefits/all-claims/tests/config/781.unit.spec.js]: no biggie, but objects is misspelled.

[the remark this answers is not in the document]
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/tests/config/781.unit.spec.js]: Thanks! Good catch.

**Comment by robfelder** [comment]
LGTM!

**Review by robfelder — APPROVED** [review]

**Review by rianfowler — APPROVED** [review]
LGTM

**Comment by djmunoz1** [comment]
@randyhoffmastergit  pls note this PR has received a review to assess adherence to basic code quality standards and does not ensure that all integration checks have passed

**Pull request merged**